pkg/crosscluster/producer/stream_lifetime.go

```diff
@@ -84,6 +84,11 @@ func StartReplicationProducerJob(
 	if tenantID.IsSystem() && !kvserver.RangefeedEnabled.Get(&evalCtx.Settings.SV) {
 		return streampb.ReplicationProducerSpec{}, errors.Errorf("kv.rangefeed.enabled must be true to start a replication job")
 	}
+	currentVersion := execConfig.Settings.Version.ActiveVersion(ctx).Version
+	if req.DestClusterVersion.Version != (roachpb.Version{}) && req.DestClusterVersion.Major < currentVersion.Major {
```

Comment:
This is tricky to do with the existing version infrastructure, but this isn't the right assertion. It's okay to replicate from a new physical cluster as long as the virtual clusters version is compatible with the destination cluster.

This could be relevant for Serverless's use of PCR for transparent cluster migration. They might move a tenant onto a new host, then want to move it back because there is a problem with the new cluster version.

Reply:
Thinking a bit more about how the version checks could work. The producer job on the source cluster could own a PTS and a version lock. The version lock would be the destination clusters version and would prevent finalizing the replicated cluster.